cvat/apps/dataset_manager/annotation.py

```diff
@@ -523,7 +524,7 @@ def match_left_right(left_curve, right_curve):
                 return matching
 
             def match_right_left(left_curve, right_curve, left_right_matching):
-                matched_right_points = left_right_matching.values()
+                matched_right_points = list(chain.from_iterable(left_right_matching.values()))
```

Comment:
I think, it may be better to use some kind of search tree here for better performance. For instance, you can create a list of `id(obj)` and put it into a `set`.